### PR TITLE
Fix Geonotebook role and add KTile role

### PR DESCRIPTION
**Pull request by danlamanna**
@kotfic Give this a shot.

`vagrant up` should give you what you want. Note this mounts your current directory as the path to the code.

The only annoyance is that in order to figure out the token, you need to ssh into the machine and type `systemctl status geonotebook`. There's a variable that disables tokens altogether, how should we handle the defaults? I can see running the status command and printing the output after provisioning the VM.

### CODE REVIEW

**Comment by kotfic** [comment]
This looks good to me once we can figure out how to communicate to the url to the user.  ideally this would be displayed as apart of the post_up_message - is there a way to run a ruby function after the machine is up?  (e.g. ```$(vagrant ssh systemctl status geonotebook)```)?

**Comment by danlamanna** [comment]
The easiest way I can think of doing it would be to add a shell provisioner after the ansible provisioner.

**Comment by danlamanna** [comment]
This turned out to be far more complicated than I wanted due to having to persist the token on the client side. Basically, the role now supports (in a cleaner way) handling auth tokens, but Vagrant disables them. I'm all for secure defaults, but the vagrant handler is only meant to be used in a development environment on localhost.

**Comment by kotfic** [comment]
I am seeing the following error:

```
TASK [ktile : Clone the ktile repository] **************************************
fatal: [default]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find required executable git"}
```

logging in reveals git is not available:

```
Welcome to Ubuntu 16.04.1 LTS (GNU/Linux 4.4.0-31-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
Last login: Mon Jan 30 15:16:19 2017 from 10.0.2.2
vagrant@geonotebook:~$ git
The program 'git' is currently not installed. You can install it by typing:
sudo apt install git
```

**Comment by kotfic** [comment]
@danlamanna Just a few details:

+ Lets add ```curl``` and ```awscli``` to requirements (so notebook examples work).  
+ Lets add ```scikit-image``` and ```seaborn``` to python requirements
+ Can we make the branch that is checked out configurable (default to master), e.g. with an environment variable
+ Can we rename the vagrant machine from ```default``` to ```geonotebook```?  
+ I'd like to make RAM/CPU's configurable and to default to 2 CPUs  and 2Gb of RAM 
+ Last can we add a Vagrant.md that documents the configurable options,  the geonotebook_auth_token,  etc,  then link it in the main README.md

**Comment by danlamanna** [comment]
@kotfic Rebased and addressed your concerns, ptal.

File: Vagrant.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+| GEONOTEBOOK_UPDATE       | Documentation for this can be found in the [Geonotebook role docs](devops/roles/geonotebook).       |
+| GEONOTEBOOK_FORCE        | Documentation for this can be found in the [Geonotebook role docs](devops/roles/geonotebook).       |
+| GEONOTEBOOK_AUTH_ENABLED | Documentation for this can be found in the [Geonotebook role docs](devops/roles/geonotebook).       |
+| GEONOTEBOOK_AUTH_TOKEN   | Documentation for this can be found in the [Geonotebook role docs](devops/roles/geonotebook).       |
+
+
+Jupyter Auth
+------------
+By default, Jupyter authentication is disabled in the Vagrant environment as it's meant to be a development environment. Jupyter authentication should always be enabled on production systems.
+
```
Comment (kotfic) [inline Vagrant.md]: 👍

**Pull request merged**